plenum/server/replica.py

```diff
@@ -2058,7 +2062,8 @@ def enqueue_prepare(self, pMsg: Prepare, sender: str):
             self.preparesWaitingForPrePrepare[key] = deque()
         self.preparesWaitingForPrePrepare[key].append((pMsg, sender))
         if key not in self.pre_prepares_stashed_for_incorrect_time:
-            self._request_pre_prepare_for_prepare(key)
+            if self.isMaster or not self.last_ordered_3pc[1] != 0:
```

Comment:
Should it be `if self.isMaster or self.last_ordered_3pc[1] != 0:`?

Reply:
This case does not relate to the issue with the last ordered key indeterminacy on backup replicas after catch-up. So I think that we should keep the existing logic.

Reply:
fixed it